core/src/Cabana_VerletList.hpp

```diff
@@ -37,23 +37,23 @@ struct VerletLayout2D
 //---------------------------------------------------------------------------//
 // Verlet List Data.
 //---------------------------------------------------------------------------//
-template <class DeviceType, class LayoutTag>
+template <class MemorySpace, class LayoutTag>
 struct VerletListData;
 
-template <class DeviceType>
-struct VerletListData<DeviceType, VerletLayoutCSR>
+template <class MemorySpace>
+struct VerletListData<MemorySpace, VerletLayoutCSR>
 {
     // Device type.
-    using device_type = DeviceType;
+    using memory_space = MemorySpace;
```

Comment:
`device_type` was a public type member so this may break user code (just making sure you are aware of this).

You could do
```C++
using device_type [[deprecated]] = typename MemorySpace::device_type;
```
to be safe.

Reply:
Also you might want to do
```C++
static_assert(Kokkos::is_memory_space<MemorySpace>::value, "");

Reply:
Good point - I re-added the `device_type` everywhere and added the `assert` below in the main class